clustered/server/src/test/java/org/ehcache/clustered/server/store/ServerStoreTest.java

```diff
@@ -0,0 +1,222 @@
+    ServerStore store = newStore();
+    Chain chain = store.get(1);
+    assertThat(chain.isEmpty(), is(true));
+    assertThat(chain.iterator().hasNext(), is(false));
+  }
+
+  @Test
+  public void testGetMappingExists() {
+    ServerStore store = newStore();
+    populateStore(store);
+    Chain chain = store.get(1);
+    assertThat(chain.isEmpty(), is(false));
+    assertChainHasOnly(chain, 1);
+    assertReverseChainHasOnly(chain, 1);
+  }
+
+  @Test
+  public void testAppendNoMappingExists() {
+    ServerStore store = newStore();
+    store.append(1, getPayload(1));
+    Chain chain = store.get(1);
+    assertThat(chain.isEmpty(), is(false));
+    assertChainHasOnly(chain, 1);
+    assertReverseChainHasOnly(chain, 1);
+  }
+
+  @Test
+  public void testAppendMappingExists() {
+    ServerStore store = newStore();
+    populateStore(store);
+    store.append(2, getPayload(22));
+    Chain chain = store.get(2);
+    assertThat(chain.isEmpty(), is(false));
+    assertChainHasOnly(chain, 2, 22);
+    assertReverseChainHasOnly(chain, 22, 2);
+  }
+
+  @Test
+  public void testGetAndAppendNoMappingExists() {
+    ServerStore store = newStore();
+    Chain chain = store.getAndAppend(1, getPayload(1));
+    assertThat(chain.isEmpty(), is(true));
+    chain = store.get(1);
+    assertChainHasOnly(chain, 1);
+    assertReverseChainHasOnly(chain, 1);
+  }
+
+  @Test
+  public void testGetAndAppendMappingExists() {
+    ServerStore store = newStore();
+    populateStore(store);
+    Chain chain = store.getAndAppend(1, getPayload(22));
+    for (Element element : chain) {
+      assertThat(readPayLoad(element.getPayload()), is(Long.valueOf(1)));
+    }
+    chain = store.get(1);
+    assertChainHasOnly(chain, 1, 22);
+    assertReverseChainHasOnly(chain, 22, 1);
+  }
+
+  @Test
+  public void testReplaceAtHeadSucceedsMappingExistsHeadMatchesStrictly() {
+    ServerStore store = newStore();
+    populateStore(store);
+    Chain existingMapping = store.get(1);
+
+    store.replaceAtHead(1, existingMapping, chainBuilder.build(compactChain(existingMapping, getPayload(11))));
+    Chain chain = store.get(1);
+    assertChainHasOnly(chain, 11);
+    assertReverseChainHasOnly(chain, 11);
+
+    store.append(2, getPayload(22));
+    store.append(2, getPayload(222));
+
+    existingMapping = store.get(2);
+
+    store.replaceAtHead(2, existingMapping, chainBuilder.build(compactChain(existingMapping, getPayload(2222))));
+
+    chain = store.get(2);
+
+    assertChainHasOnly(chain, 2222);
+    assertReverseChainHasOnly(chain, 2222);
+  }
+
+  @Test
+  public void testReplaceAtHeadSucceedsMappingExistsHeadMatches() {
+    ServerStore store = newStore();
+    populateStore(store);
+
+    Chain existingMapping = store.get(1);
+
+    store.append(1, getPayload(11));
+
+    store.replaceAtHead(1, existingMapping, chainBuilder.build(compactChain(existingMapping, getPayload(111))));
+    Chain chain = store.get(1);
+
+    assertChainHasOnly(chain, 111, 11);
```

Comment:
You always combine this assert and the one on the next line - it could be extracted in a single method.